### ISSUE TITLE
Data node stops processing new blocks during overnight run

### ISSUE THREAD

**Issue by jgsbennett** [vegaprotocol/vega#8185]
# Problem encountered
During last night's nightly run, the data node's core node seems to have stopped processing new blocks for some reason, causing all remaining tests to fail.
```
2023-04-20T02:02:44.189Z	INFO	tendermint	consensus/ticker.go:124	Timed out	{"dur": "3s", "height": 5957, "round": 0, "step": "RoundStepPropose"}
2023-04-20T02:09:58.593Z	INFO	tendermint	conn/connection.go:600	Connection is closed @ recvRoutine (likely by the other side)	{"conn": "MConn{127.0.0.1:26606}"}
```
Note the 7 minute delay between the last block and the next message, which is when all the nodes start getting closed down by Jenkins. During the same time, the core nodes were happily ticking along with new blocks, but this is not reflected in the data node's core node logs.

# Observed behaviour
No new events received from data node, since it has stopped processing blocks

# Expected behaviour
Data node is producing blocks as long as core node does.

# System response
Lots of tests failed:
https://jenkins.ops.vega.xyz/job/common/job/system-tests-nightly/902/testReport/
In the exact case, the first test in that run was doing some spam things. Specifically, it's in the middle of making a network proposal to change some proof of work stuff. Seemingly it opens a proposal, votes on it, then the tests get very sad because they never see any events for the vote or enacted proposal, and that's because the data node has stopped seeing new blocks.
The rest of the run then bails because the data node is broken, so all failures after that first one can be ignored. 

# Steps to reproduce

## Automation
Unknown. Seen only once so far. I'm working to see if I can reproduce.

# Evidence

## Logs
[data node logs.zip](https://github.com/vegaprotocol/vega/files/11286694/data.node.logs.zip)
[core logs.zip](https://github.com/vegaprotocol/vega/files/11286695/core.logs.zip)

## Additional context
Add any other context about the problem here including; system version numbers, components affected.

# Definition of Done
>ℹ️ Not every issue will need every item checked, however, every item on this list should be properly considered and actioned to meet the [DoD](https://github.com/vegaprotocol/vega/blob/develop/DEFINITION_OF_DONE.md).

**Before Merging**
- [ ] Code refactored to meet SOLID and other code design principles
- [ ] Code is compilation error, warning, and hint free
- [ ] Carry out a basic happy path end-to-end check of the new code
- [ ] All APIs are documented so auto-generated documentation is created
- [ ] All bug recreation steps can be followed without presenting the original error/bug
- [ ] All Unit, Integration and BVT tests are passing
- [ ] Implementation is peer reviewed (coding standards, meeting acceptance criteria, code/design quality)
- [ ] Create [front end](https://github.com/vegaprotocol/token-frontend/issues) or [console](https://github.com/vegaprotocol/console/issues) tickets with feature labels (should be done when starting the work if dependencies known i.e. API changes)

**After Merging**
- [ ] Move development ticket to `Done` if there is **NO** requirement for new [system-tests](https://github.com/vegaprotocol/system-tests/issues)
- [ ] Resolve any issues with broken system-tests
- [ ] Create [documentation](https://github.com/vegaprotocol/documentation/issues) tickets with feature labels if functionality has changed, or is a new feature

**Comment by pscott31** [comment]
I don't _think_ this is a data node problem per se. What seems to have happened is 

02:02 - Data node stopped getting events from core:

```2023-04-20T02:02:42.690Z	WARN	datanode.sqlstore-broker	broker/sqlstore_broker.go:131	slow time update detected, time between checks 2s, block height: 5956, total block processing time: 5.018851ms```

At this point in time core's log goes earily silent until 02:09, which is highly unusual - it is ordinarily extremely chatty, so I think it is blocked somehow (gc??)

02:09 - Core node is sent a SIGINT or SIGTERM by something (nomad??)
```2023-04-20T02:09:58.617Z	INFO	core	node/node.go:174	Caught signal	{"name": "interrupt"}```

And immediately data node notices:

```2023-04-20T02:09:58.607Z	INFO	datanode.broker.eventsource.socket-server	broker/socket_server.go:90	New broker connection event	{"eventType": "Detached", "id": 891240226, "address": "tcp://[::]:33025"}```

**Comment by gordsport** [comment]
This issue is not presenting any more, closing for now. To be reopened (or a new issue created) if we see this issue present again.
